File: cpp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/pendientes/E02_RepasoBucleSuma_Pendiente.cpp
/*
OBJETIVO: For: suma del 1 al 10.
PATRON: Menu do-while con opcion 0 salir y uso de cin.
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E02 Repaso Bucle Suma Pendiente ===" << endl;
    cout << "1. Revisar el ejercicio guiado" << endl;
    cout << "0. Salir" << endl;
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                cout << "Prepara una variable acumuladora y un bucle for del 1 al 10." << endl;
                cout << "Al final debes mostrar la suma total." << endl;
                cout << "Pista: empieza con s = 0." << endl;
                // TODO: completar el bucle for que suma del 1 al 10.
                break;
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}

File: csharp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/pendientes/E02_RepasoBucleSuma_Pendiente.cs
/*
OBJETIVO: For: suma del 1 al 10.
PATRON: Menu do-while con opcion 0 salir y uso de Console.ReadLine().
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/
using System;
class Program
{
    static void Main()
    {
        string? opcion;
        do
        {
            Console.WriteLine();
            Console.WriteLine("=== E02 Repaso Bucle Suma Pendiente ===");
            Console.WriteLine("1. Revisar el ejercicio guiado");
            Console.WriteLine("0. Salir");
            Console.Write("Elige una opcion: ");
            opcion = Console.ReadLine();
            switch (opcion)
            {
                case "1":
                    Console.WriteLine("Prepara una variable acumuladora y un bucle for del 1 al 10.");
                    Console.WriteLine("Al final debes mostrar la suma total.");
                    Console.WriteLine("Pista: empieza con s = 0.");
                    break;
                case "0":
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida.");
                    break;
            }
        } while (opcion != "0");
    }
}
